example/osmpbf/main.rs

```diff
@@ -1,54 +1,63 @@
-                }
-                Element::Relation(_rel) => {
-                    // do nothing
-                }
-                Element::Node(node) => {
-                    let point = (node.lon(), node.lat());
-                    nodes.insert(node.id(), point);
-                    let tags = node.tags().into_iter().collect();
-                    let encoded = encode::node(node.id() as u64, point, tags).unwrap();
-                    println!("{}", hex::encode(encoded));
-                }
-                Element::Way(way) => {
-                    for r in way.refs() {
-                        let ref item = nodes[&r];
-                        deps.entry(r).or_insert(*item);
-                    }
-                    let tags = way.tags().into_iter().collect();
-                    let refs = way.refs().into_iter().collect();
-                    let encoded = encode::way(way.id() as u64, tags, refs, &deps).unwrap();
-                    println!("{}", hex::encode(encoded));
-                }
+    reader.for_each(|item| {
+        match item {
+            Element::DenseNode(node) => {
+                let point = (node.lon(), node.lat());
+                nodes.insert(node.id, point);
+                let tags = node.tags().into_iter().collect();
+                let encoded = encode::node(node.id as u64, point, &tags).unwrap();
+                println!("{}", hex::encode(encoded));
+            },
+            Element::Node(node) => {
+                let point = (node.lon(), node.lat());
+                nodes.insert(node.id(), point);
+                let tags = node.tags().into_iter().collect();
+                let encoded = encode::node(node.id() as u64, point, &tags).unwrap();
+                println!("{}", hex::encode(encoded));
+            },
+            Element::Relation(rel) => {
+                let tags = rel.tags().into_iter().collect();
+                let members = rel.members().map(|m| convert_member(&m)).collect();
+                let encoded = encode::relation(rel.id() as u64, &tags, &members, &nodes, &ways).unwrap();
+                println!("{}", hex::encode(encoded));
+            },
+            Element::Way(way) => {
+                let tags = way.tags().into_iter().collect();
+                let refs: Vec<i64> = way.refs().into_iter().collect();
+                ways.insert(way.id(), refs.clone());
+                let encoded = encode::way(way.id() as u64, &tags, &refs, &nodes).unwrap();
+                println!("{}", hex::encode(encoded));
             }
-        })
-        .unwrap();
-
+        }
+    }).unwrap();
     Ok(())
 }
+
+fn convert_member(m: &RelMember) -> Member {
+    Member::new(
+        m.member_id as u64,
+        match m.role().unwrap() {
+            "inner" => MemberRole::Inner(),
+            "outer" => MemberRole::Outer(),
+            _ => MemberRole::Unused(),
+        },
+        match m.member_type {
+            RelMemberType::Node => MemberType::Node(),
```

Comment:
This makes sense but also was initially confusing because `MemberRole` & `MemberType` is defined in this crate and `RelMemberType` is part of `osmpbf`. In other typed languages, the convention is to use the package name if it would be non-obvious, so it would change to

```
osmpbf::RelMemberType::Node => MemberType::Node(),
osmpbf::RelMemberType::Way => MemberType::Way(),
osmpbf::RelMemberType::Relation => MemberType::Relation(),
```